contracts/registries/LicenseRegistry.sol

```diff
@@ -430,10 +440,14 @@ contract LicenseRegistry is ILicenseRegistry, AccessManagedUpgradeable, UUPSUpgr
         }
         // childIp can only register with default license terms or the license terms attached to the parent IP
         if ($.defaultLicenseTemplate != licenseTemplate || $.defaultLicenseTermsId != licenseTermsId) {
-            if ($.licenseTemplates[parentIpId] != licenseTemplate) {
+            address pLicenseTemplate = $.licenseTemplates[parentIpId];
+            if (
+                (isUsingLicenseToken && pLicenseTemplate != address(0) && pLicenseTemplate != licenseTemplate) ||
```

Comment:
Is `pLicenseTemplate != address(0)` duplicate of `pLicenseTemplate != licenseTemplate`? Or is `licenseTemplate == address(0)` true sometimes so we need to check for `pLicenseTemplate != address(0)`?

Reply:
yes, when parent IP has already attached license (`pLicenseTemplate == address(0)`) then the `licenseTemplate == pLicenseTemplate` should be true.  when parent IP has not yet attach any license (`pLicenseTemplate == address(0)`)  as it is private license, in that case `licenseTemplate != pLicenseTemplate` should be true.